magento/inventory
# MSI-1381 // SalesChannel validation for API requests

### Description
This PR provides additional api-functional testing for SalesChannel to Stock assignment and extends the api-functional tests for Validation. Further it extends the validation chain.

### Fixed Issues (if relevant)
1. magento-engcom/msi#1381 : There is no validation or fallback value for "type" property of the "sales_channels" extension attribute in "save" method of Magento/InventoryApi/Api/StockRepositoryInterface

### Manual testing scenarios
This PR can be easy tested by following the instructions in the issue https://github.com/magento-engcom/msi/issues/1381 
1. Execute the provided script (copy it into a file and update the three variables `endpoint`, `username` and `password`)
2. After execution the provided script the given sales channel assignments should still exits
3. Exception message describes the error

You can also test this PR by running the api-functional tests for REST and SOAP

### Contribution checklist
